api/src/main/java/io/minio/messages/TopicConfiguration.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Minio Java SDK for Amazon S3 Compatible Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.minio.messages;
+
+import java.util.LinkedList;
+import java.util.List;
+
+import org.xmlpull.v1.XmlPullParserException;
+
+import com.google.api.client.util.Key;
+
+
+/**
+ * Helper class to parse Amazon AWS S3 response XML containing topic configuration.
+ */
+public class TopicConfiguration extends XmlEntity {
+  @Key("Topic")
+  private String topic;
+  @Key("Event")
+  private List<String> events = new LinkedList<>();
+  @Key("Filter")
+  private Filter filter;
+
+
+  public TopicConfiguration() throws XmlPullParserException {
+    super();
+    super.name = "TopicConfiguration";
+  }
+
+
+  /**
+   * Returns topic.
+   */
+  public String topic() {
+    return topic;
+  }
+
+
+  /**
+   * Sets topic.
+   */
+  public void setTopic(String topic) {
+    this.topic = topic;
+  }
+
+
+  /**
+   * Returns events.
+   */
+  public List<String> events() {
+    return events;
+  }
+
+
+  /**
+   * Sets events.
+   */
+  public void setEvents(List<String> events) {
+    this.events = events;
+  }
+
+
+  /**
+   * Returns filter.
+   */
+  public Filter filter() {
+    return filter;
+  }
+
+
+  /**
+   * Sets filter.
+   */
+  public void setFilter(Filter filter) {
```

Comment:
> Anyhow I will try how Amazon S3 behaves and add accordingly.

Yes, it would be better to test again, Thanks. I am relying on my memory and minio-go code (https://github.com/minio/minio-go/blob/master/bucket-notification.go#L107)